### PR TITLE
[update] Work on e2e refactoring to Java - datamapper

### PR DESCRIPTION
**Pull request by Stefan365**
work on integrations/datamapper 
+ update of connections structure.
+ apply philosophy: wait for element (...shouldBe(...)) just immediately before element usage.

### CODE REVIEW

File: ui-tests/src/test/java/io/syndesis/qe/pages/SyndesisPageObject.java [hunk 1]
```diff
@@ -107,6 +107,17 @@ public SelenideElement getElementByClassName(String elementClassName) {
 		return this.getRootElement().find(By.className(elementClassName));
 	}
 
+	public SelenideElement getElementByLocator(By elementLocator) {
+		log.info(String.format("searching for element %s", elementLocator.toString()));
```
Comment (dsimansk) [inline ui-tests/src/test/java/io/syndesis/qe/pages/SyndesisPageObject.java]: You don't need String.format in log message, just `log.info("searching for element {}", elementLocator.toString());`

File: ui-tests/src/test/java/io/syndesis/qe/pages/integrations/detail/DataMapperComponent.java [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+		log.info("datamapper has been loaded");
+		ElementsCollection dmColumns = this.getRootElement().findAll(Element.DM_COLLUMNS).shouldBe(size(2));
+		log.info(String.format("found %d datamapper columns", dmColumns.size()));
+		return dmColumns;
+	}
+
+	/**
+	 * Eventually returns count of found datamapper fields.
+	 *
+	 * @returns {wdpromise.Promise<boolean>}
```
Comment (dsimansk) [inline ui-tests/src/test/java/io/syndesis/qe/pages/integrations/detail/DataMapperComponent.java]: Javadocs need a bit of refactoring too :)

File: ui-tests/src/test/java/io/syndesis/qe/pages/integrations/detail/DataMapperComponent.java [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+		log.info(String.format("source has %d fields.length fields", fields.size()));
+
+		SelenideElement nextField;
+		for (String p : path) {
+			nextField = this.findFieldByName(p, fields);
+			// click on it to expand or select find correct field from list
+			log.info(String.format("Clicking on field %s", p));
+			nextField.$(Element.LABEL).shouldBe(visible).click();
+			// find all subfields for next iteration
+			fields = nextField.$$(Element.FIELD_DETAIL).shouldBe(sizeGreaterThanOrEqual(0));
```
Comment (lvydra) [inline ui-tests/src/test/java/io/syndesis/qe/pages/integrations/detail/DataMapperComponent.java]: Isn't this shouldBe always true?

File: ui-tests/src/test/java/io/syndesis/qe/pages/integrations/detail/DataMapperComponent.java [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+		public static final By FIRST_COMBINE_POSITION = By.xpath("(//mapping-field-action//label[text()='Index']/following-sibling::input)[1]");
+		public static final By SECOND_COMBINE_POSITION = By.xpath("(//mapping-field-action//label[text()='Index']/following-sibling::input)[2]");
+		public static final By ACTION_SELECT = By.xpath("//label[text()='Action']/following-sibling::select");
+		public static final By SEPARATOR_SELECT = By.xpath("//label[text()='Separator:']/following-sibling::select");
+		public static final By TRANSFORMATION_SELECT = By.xpath("//label[text() = 'Transformation']/following-sibling::select");
+	}
+
+	@Override
+	public SelenideElement getRootElement() {
+		return $(Element.ROOT_1).$(Element.ROOT_2);
```
Comment (lvydra) [inline ui-tests/src/test/java/io/syndesis/qe/pages/integrations/detail/DataMapperComponent.java]: I'm not sure if this is suitable solution.

File: ui-tests/src/test/java/io/syndesis/qe/pages/integrations/detail/DataMapperComponent.java (hunk 3 above)
Comment (Stefan365) [reply ui-tests/src/test/java/io/syndesis/qe/pages/integrations/detail/DataMapperComponent.java]: this is here due to waiting for all elements of collection have been collected. after discussion I Updated it to:... (1)

File: ui-tests/src/test/java/io/syndesis/qe/pages/connections/list/ConnectionsListComponent.java [hunk 5]
```diff
@@ -46,54 +46,54 @@ public boolean validate() {
...
 
 	public void clickOnAllKebabButtons() throws Exception {
-		ElementsCollection kebabButtons = this.getAllKebabButtons();
-		for (SelenideElement kebabB : kebabButtons) {
-			log.debug(String.format("clicking on kebab button %s", kebabB.getText()));
-			kebabB.click();
+		ElementsCollection kebabButtons = this.getAllKebabButtons().shouldBe(sizeGreaterThanOrEqual(1));
+		for (SelenideElement kebabButton : kebabButtons) {
+			log.debug("clicking on kebab button {}", kebabButton.getText());
+			kebabButton.click();
```
Comment (lvydra) [inline ui-tests/src/test/java/io/syndesis/qe/pages/connections/list/ConnectionsListComponent.java]: shouldBe(visible)

File: ui-tests/src/test/java/io/syndesis/qe/pages/integrations/detail/DataMapperComponent.java [hunk 6]
```diff
@@ -0,0 +1,206 @@
...
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * Created by sveres on 11/14/17.
+ */
+@Slf4j
+public class DataMapperComponent extends SyndesisPageObject {
+
+	private static final class Element {
+		public static final By ROOT = By.cssSelector("syndesis-data-mapper-host > div > data-mapper");
```
Comment (lvydra) [inline ui-tests/src/test/java/io/syndesis/qe/pages/integrations/detail/DataMapperComponent.java]: Cant it be only By.cssSelector("data-mapper") it is basically selecting the same element.

Comment (Stefan365) [reply ui-tests/src/test/java/io/syndesis/qe/pages/integrations/detail/DataMapperComponent.java]: yes, it can!

**Comment by dsimansk** [comment]
@lvydra any more comments to latest code changes?

**Pull request merged**